### ISSUE TITLE
Cannot setup a breakpoint from debuggers (because of transpiler ?)

### ISSUE THREAD

**Issue by EricBoix** [iTowns/itowns#934]
## Your Environment
* Version used: V2.5.0 (git tag), SHA1 326722f83b79477a53718c78aa4b19d127f1f15c
* Browser Name and version: Firefox 63.0.3, Chrome 70.0.3538.110, Webstorm 2018.3
* Operating System and version (desktop or mobile): macOS 10.13.6
* Link to your project: [THE official iTowns](https://github.com/iTowns/itowns)

## Context
When trying to debug iTowns internal, I seem **unable to set an effective breakpoint** (in the portions of the code that are webpacked as opposed to an entry point files located in examples).
This prevents the usage of debuggers like Firefox, Chrome or Webstorm.

When inquiring about this difficulty within the Webstorm community, it appeared that a [possible reasons preventing](https://intellij-support.jetbrains.com/hc/en-us/community/posts/360001788680-Webpack-Dev-Server-debugging) the "debuggers" for running comes from the fact that the bundled code it transpiled by babel. Debuggers' Source mappers are then fooled by this transpilation.

The general problem folds back to the following user story: 
   As an iTowns developper I want to use a debugger on iTowns internal.

What is the proper, recommended setup to do so ? 

## Steps to Reproduce (for bugs)
```
git clone https://github.com/iTowns/itowns
cd itowns
npm install
npm start
```
 - Open say Firefox on `http://localhost:8082/examples/planar.html`
 - Open Web-Developper ---> Debugger tool (F12)
 - Use `%P` to search for file `WMSSource.js`. Firefox provides two possible completions:
    * `webpack-internal:///./src/Source/WMSSource.js` 
        for which the first lines go 
        ```
          __webpack_require__.r(__webpack_exports__);
          /* harmony import */ var _babel_runtime_helpers_classCallCheck__WEBPACK_IMPORTED_MODULE_0__ = ....
        ```
    * `webpack://[name]/src/Source/WMSSource.js?d8f6`
       for which the first lines go
       ```
       import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
       import _createClass from "@babel/runtime/helpers/createClass";
       ```
Hence I cannot seem to retrieve [`src/Source/WMSSource.js`](https://github.com/iTowns/itowns/blob/master/src/Source/WMSSource.js) (in order to set a break point) that starts with
```
import Source from 'Source/Source';
import URLBuilder from 'Provider/URLBuilder';
```

## Expected Behavior
Be able to set breakpoint with the debugger.

## Actual Behavior
Unable to set a breakpoints with the debugger.

## Possible Cause/Fix/Solution
Using Webstorm as debugger, I tried to inhibit babel transpiling and fooled around with
 - webpack cli options (removing all of them) as found in [package.json](https://github.com/jailln/itowns/blob/master/package.json#L16)
 - the partial or complete removal of [.babelrc](https://github.com/jailln/itowns/blob/master/.babelrc) (which most often produces webpack failures)
 - babel excisions out of [webpack.config.js](https://github.com/jailln/itowns/blob/master/webpack.config.js#L74) (but this also breaks webpack run)

When lucky I get to prevent Webstorm from [popping a webpack:[name]... file tab](https://intellij-support.jetbrains.com/hc/en-us/community/posts/360001788680-Webpack-Dev-Server-debugging) but without being able to have effective breakpoints.

**Comment by gchoqueux** [comment]
Thank you for reporting this problem, it is actually an important user story.

For my part, I had no problem with debugging.

To find the solution, I need some clarification:
* ~Are you running a server with the command `npm start`?~
* Do you add the breakpoint in webStorm **or** in Chrome / FireFox?
* Have you try adding the `debugger;` directly in the itowns code?

**Comment by EricBoix** [comment]
Thanx @gchoqueux . To answer your questions 
* I edited the issue to be more precise on the reproducing steps
* I changed the issue title to be more precise too
* I first tried adding a breakpoint directly from/with the two tools I usually use for "quick" debug i.e. Firefox or Chrome (I did that in separate contexts i.e. Firefox alone OR Chrome Alone).
* Because I could only access transpiled code (which is normal because this is the only thing you seem to get when opening the URL), I tried using Webstorm (that I configured to trigger Firefox on entering debug) for which you can provide the actual (iTowns) sources on condition of also providing the "source mapping files". I then thought I was doing something wrong with WebStorm (this is why I [opened an "issue" on their site](https://intellij-support.jetbrains.com/hc/en-us/community/posts/360001788680-Webpack-Dev-Server-debugging)) to which I'm newbie.
* Adding a `debugger;` line in the itowns source (at the ad-hoc line) does indeed work as an equivalent for setting up a breakpoint. I guess this can be considered as a decent/acceptable workaround although one ends up (as expected) in the transpiled code. If the debugged section is not too much transformed by the transpiler then this does the job.

**Comment by EricBoix** [comment]
## Currently adopted work around:
 - edit iTowns source at your ad-hoc spot and add the following two lines (and yes you **do** need the comment line)
   ```
    // eslint-disable-next-line no-debugger
    debugger;
   ```
 - the debugger will break where required ... up to the fact that you will end up in (babel) transpiled code (which might be confusing depending on the section you wish to debug).

**Comment by gchoqueux** [comment]
I have identified the problem, it comes from a babel plugin `babel-inline-import-loader`.

**Temporary workaround**

1. Comment this [line to disable `babel-inline-import-loader`](https://github.com/iTowns/itowns/blob/master/webpack.config.js#L74) 

**Step to reproduce**

2. run `npm start`
3. Open say Firefox on `http://localhost:8080`
4. Open Web-Developper ---> Debugger tool (F12)
5. Use `%P` or `ctrl + p` to search for file `WMSSource.js`.
    * use the one that starts with `webpack://[name]/`
       for which the first lines go
       ```js
       import Source from 'Source/Source';
       import URLBuilder from 'Provider/URLBuilder';
       ```

**Comment by EricBoix** [comment]
Thanks @gchoqueux .
Wonderful.  This does the trick for me with both Firefox and Webstorm as debuggers ! (I didn't check it with Chrome but it should work accordingly). 
Maybe this information should end up in the developers guide somewhere ?
Anyhow, and IMHO, the above provided solution solves the issue. Should I be the one closing it ?

**Comment by gchoqueux** [comment]
fix in #1144
Now,  run `run npm debug`
